python_modules/dagster/dagster/_utils/schedules.py

```diff
@@ -1,14 +1,24 @@
 import datetime
+import functools
 from typing import Iterator, Optional, Sequence, Union
 
 import pendulum
 import pytz
-from croniter import croniter
+from croniter import croniter as _croniter
 
 import dagster._check as check
 from dagster._seven.compat.pendulum import to_timezone
 
 
+class croniter(_croniter):
+    """Lightweight shim to enable caching certain values that may be calculated many times."""
```

Comment:
this feels a little magical to me - consider something like CroniterShim or something (not blocking)